cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -72,7 +72,7 @@ class BufferExpander
     BufferExpander(std::shared_ptr<ConcreteBuffer> const & buf, bool clone, ctor_passkey const &)
         : m_concrete_buffer(clone ? buf->clone() : buf)
         , m_begin(m_concrete_buffer->data())
-        , m_end(m_begin + size())
+        , m_end(m_begin + m_concrete_buffer->size())
```

Comment:
`m_end` should use `m_concrete_buffer->size()`.  It is wrong to call `size()`, which reads `m_end` at line 96!

```cpp
    size_type size() const noexcept
    {
        return static_cast<size_type>(this->m_end - this->m_begin);
    }
```